scripts.js

```diff
@@ -0,0 +1,244 @@
+  firstName: 'Adam',
+  lastName: 'Johnson'
+}
+
+console.log(checkIfUsersHaveTheSameName(firstJohn, secondJohn)); // true
+console.log(checkIfUsersHaveTheSameName(firstJohn, adam)); // false
+
+// 6
+function getVoteCount(votes) {
+  return votes.upvotes - votes.downvotes;
+}
+
+console.log(getVoteCount({ upvotes: 10, downvotes: 5 })); // 5
+console.log(getVoteCount({ upvotes: 75, downvotes: 90 })); // -15
+console.log(getVoteCount({ upvotes: 50, downvotes: 50 })); // 0
+
+// 7
+function getCubeVolume(cubeDimensions) {
+  return cubeDimensions.width * cubeDimensions.height * cubeDimensions.length;
+}
+
+console.log(getCubeVolume({ width: 10, length: 5, height: 2 })); // 100
+console.log(getCubeVolume({ width: 100, length: 500, height: 0 })); // 0
+console.log(getCubeVolume({ width: 15, length: 2, height: 5 })); // 150
+
+// 8
+function getCityInformation(cityObject) {
+  return `${cityObject.name} is in ${cityObject.country} and has an area of ${cityObject.areaInKilometers} square kilometers`;
+}
+
+console.log(getCityInformation({
+  name: 'Warsaw',
+  country: 'Poland',
+  areaInKilometers: 517
+})); // Warsaw is in Poland and has an area of 517 square kilometers
+
+// 9
+function getUserCopy(user) {
+  return {
+    firstName: user.firstName,
+    lastName: user.lastName,
+  };
+}
+
+const john = {
+  firstName: 'John',
+  lastName: 'Smith'
+}
+
+const newUser = getUserCopy(john);
+
+console.log(newUser.firstName); // John
+console.log(newUser.lastName); // Smith
+console.log(newUser === john); // false <-- this is crucial
+
+//10
+function getAnimalDescription(animalObject) {
+  return `This ${animalObject.color} ${animalObject.name} has ${animalObject.legs} legs`;
+}
+
+const animal = {
+  name: "dog",
+  legs: 4,
+  color: "white",
+}
+
+console.log(getAnimalDescription(animal));
+
+//11
+const rooms = {
+  first: {
+    description: 'This is the first room',
+    items: {
+      chair: 'The old chair looks comfortable',
+      lamp: 'This lamp looks ancient',
+    }
+  },
+  second: {
+    description: 'This is the second room',
+    items: {
+      couch: 'This couch looks like it would hurt your back',
+      table: 'On the table there is an unopened bottle of water',
+    }
+  },
+}
+
+//12
+function addContact(object, contactName, contactPhone) {
+  const newContacts = Object.assign({}, object);
+
+  newContacts[contactName] = {
+    name: contactName,
+    phoneNumber: contactPhone
+  }
+
+  return newContacts;
+}
+
+
+const contacts = {};
+const contactsWithJohn = addContact(contacts, 'John', '123 456 789');
+
+console.log(contactsWithJohn.John.name); // "John"
+console.log(contactsWithJohn.John.phoneNumber); // '123 456 789'
+
+console.log(contacts === contactsWithJohn); // false
+console.log(contacts.John); // undefined
+
+const contactsWithKate = addContact({}, 'Kate', '789 123 456');
+console.log(contactsWithKate.Kate.name); // Kate
+
+//13
+function findContact(objectName, searchedName) {
+  return objectName[searchedName];
```

Comment:
This is not how the function was supposed to work. Please read the description again